lib/utils.js

```diff
@@ -53,13 +53,29 @@ export function addressMatch(a, b) {
 
 	// TODO comprehensive error handling and logging! This can help find issues pretty fast
 
-	// The canonical form is just the hex-portion of the address
-	// without an eventual starting 0x...
-	const canonicalA = a.toLowerCase().replace(/^0x/, "") 
-	const canonicalB = b.toLowerCase().replace(/^0x/, "")
+	// The canonical form is a fixed length, lower-case hex string
+	const canonicalA = fixedLengthHexString(a, 40) // a.t.m. ethereum has 20 Byte addresses, zero-pad it
+	const canonicalB = fixedLengthHexString(b, 40)
 
 	const isMatch = canonicalA == canonicalB
+	//console.log(`Address-Match ${a} (${canonicalA}) <> ${b} (${canonicalB})? ${isMatch} `)
 
-	console.log(`Address-Match ${a} (${canonicalA}) <> ${b} (${canonicalB})? ${isMatch} `)
 	return isMatch
 }
+
+export function fixedLengthHexString(address, length) {
+	// Returns a lower-case hex string in Format 0x{length}
+	const hexPart = address.toLowerCase().replace(/^0x/, "")
+	const paddedHexPart = hexPart.padStart(length, '0')
+	return `0x${paddedHexPart}`
+}
+
+export function nftDefined(nft) {
```

Comment:
Do you think we should implement this logic elsewhere? Also, having an NFT is basically checking whether it contains metadata or an active asset, is that assumption correct? I've been wanting to unify fetching NFTs into a single service (or custom query method) and I wonder if we should move this logic there.

Reply:
Yes, correct and yes, we can move the logic there [or keep it in utils and use it from there]. Keeping it in utils allows and moreover external developers to use it from different parts.